src/main/java/com/suse/saltstack/netapi/event/EventStream.java

```diff
@@ -129,99 +126,107 @@ public void removeEventListener(EventListener listener) {
     }
 
     /**
-     * Helper method that returns the current number of subscribed
-     * listeners.
+     * Helper method that returns the current number of subscribed listeners.
+     *
      * @return The current number listeners.
      */
     public int getListenerCount() {
-        return listeners.size();
+        int size = 0;
+        synchronized (listeners) {
+            size = listeners.size();
+        }
+        return size;
+    }
+
+    /**
+     * Helper method to check if the WebSocket Session exists and is open.
+     *
+     * @return A flag indicating the {@link EventStream}
+     * WebSocket {@link Session} state.
+     */
+    public boolean isEventStreamClosed() {
```

Comment:
```java
return this.session == null || !this.session.isOpen();
```
feels easier to read "the event stream is closed if session is null or session is not open."